lib/tools/android-manifest.js

```diff
@@ -68,42 +154,29 @@ manifestMethods.processFromManifest = async function (localApk) {
  *                 application package.
  */
 manifestMethods.packageAndLaunchActivityFromManifest = async function (localApk) {
-  try {
-    await this.initAapt();
-    log.info("Extracting package and launch activity from manifest");
-    let args = ['dump', 'badging', localApk];
-    let {stdout} = await exec(this.binaries.aapt, args);
-    let apkPackage = new RegExp(/package: name='([^']+)'/g).exec(stdout);
-    if (apkPackage && apkPackage.length >= 2) {
-      apkPackage = apkPackage[1];
-    } else {
-      log.errorAndThrow(`Cannot parse package name from ` +
-        `'${_.join([this.binaries.aapt, 'dump', 'badging', '"' + localApk + '"'], ' ')}' command  output`);
-    }
-    let apkActivity = new RegExp(/launchable-activity: name='([^']+)'/g).exec(stdout);
-    if (apkActivity && apkActivity.length >= 2) {
-      apkActivity = apkActivity[1];
-    } else {
-      let outputPath = path.resolve(this.tmpDir, apkPackage);
-      let getLaunchActivity = ['-jar', this.jars['appium_apk_tools.jar'],
-                               'printLaunchActivity', localApk,
-                               outputPath];
-      let {stdout, stderr} = await exec('java', getLaunchActivity);
-      if (stderr) {
-        log.errorAndThrow(`Cannot parse launchActivity from manifest: ${stderr}`);
-      }
-      let act = new RegExp(/Launch activity parsed:([^']+)/g).exec(stdout);
-      if (act && act.length >= 2) {
-        apkActivity = act[1];
+  const apkInfoGetters = [
+    async () => await extractApkInfoWithApkanalyzer(localApk),
+    async () => await extractApkInfoWithApkTools(localApk),
+  ];
+
+  let savedError;
+  for (const infoGetter of apkInfoGetters) {
+    try {
+      const {apkPackage, apkActivity} = await infoGetter();
+      log.debug(`Package name: ${apkPackage}`);
```

Comment:
I think the values ought to be in single quotes (i.e., `Package name: '${apkPackage}'`), in case the tools return something weird that doesn't get printed right.